src/session/session_dhandle.c

```diff
@@ -509,6 +511,22 @@ __wt_session_get_btree_ckpt(WT_SESSION_IMPL *session, const char *uri, const cha
     do {
         ret = 0;
 
+        /*
+         * Save the checkpoint generation number and the checkpoint's state to detect races.
+         * Note that we need to save the generation number before the checkpoint's state. Indeed, if
+         * we save the generation number after, we could have the following scenario:
+         *
+         * 1) The checkpoint's state is evaluated to false.
+         * 2) -- Race, checkpoint starts and finishes.
```

Comment:
The Race must be, checkpoint starts and updates the generation number. If it finishes here, there shouldn't be any problem accessing different checkpoint versions.

Reply:
That's correct, my bad! Fixed in 06d85b8fd04838c05c851c4df189a141363259e1.